2025/lab/Keys/Week8_Binary_Outcomes.Rmd

```diff
@@ -0,0 +1,406 @@
+---
+title: "Week 8 -  Regression with Binary Outcomes"
+author: "Brent Hoagland, Lab TA"
+date: "3/24/2025"
+output:
+  pdf_document: default
+  html_document: default
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+options(scipen = 10)
+
+library(tidyverse)
+library(rstanarm)
+library(modelsummary)
+library(broom.mixed)
+options(modelsummary_get = "broom") 
+library(flextable)
+library(naniar)
+
+seed <- 1234
+```
+
+# Data Loading and Management
+
+Let's use the GSS again. This week, we'll be investigating factors associated with whether a person supports marijuana legalization. 
+
+```{r gss-loading}
+# You may need to write the code or change your working directory to properly load the GSS2022.dta file and assign it to the object gss2022
+gss2022 <- read_dta("../DATASETS/GSS/GSS2022.dta")
+
+gss2022$ballot
+
+gss <- gss2022 %>% 
+  select(grassnv,                           # Target: Whether marijuana should be legal (1 = yes, 2 = no)  
+                                          # https://gssdataexplorer.norc.org/variables/285/vshow
+         polviews,                        # Political Vars 
+         sex, race, hispanic, age, educ,  # Other Demos
+         wtssnrps, ballot                    # Weight & Ballot
+         ) %>% 
+  haven::zap_labels() %>% 
+  mutate( 
+    
+    legal_grass = case_when(
+      grassnv == 1 ~ 1,
+      grassnv == 2 ~ 0
+    ),
+    conservatism = polviews, # Rennaming to be more descriptive 
+
+    age = case_when(
+      age > 88 ~ NaN,
+      TRUE ~ age
+      ),
+    sex = factor(sex, levels = c(1,2),
+                 labels = c("Male", "Female")
+                 ),
+    race = case_when(
+      hispanic != 1 ~ "Hispanic",
+      race == 1 & hispanic == 1 ~ "White",
+      race == 2 & hispanic == 1 ~ "Black",
+      race == 3 & hispanic == 1 ~ "Other",
+      ),
+    race4 = factor(race, levels = c("White", "Black", "Hispanic", "Other")),
+    weight =  wtssnrps 
+    )  %>%   
+  filter(ballot != 1) %>% 
+  select(-wtssnrps , -hispanic, -race, -polviews, -grassnv, -ballot)
+
+
+```
+
+Let's take a look at our two new variables. 1) `legal_grass`, which asks whether marijuana should be legalized; 2) `conservatism`, which is a 1 to 7 scale of ideology, with 1 as extremely liberal and 7 as extremely conservative. Let's double check the coding below. 
```

Comment:
I'd recommend grouping this into three category categorical rather than using al 7 (and certainly would recommend against treating it as linear - not totally sure whether this is what is happening).